test/TestConfig/scripts/tools/getDependencies.pl

```diff
@@ -99,6 +131,16 @@
 				die "ERROR: downloading $url failed, return code: $? \n";
 			}
 		}
+
+		# validate dependencies sha1 sum
+		$sha->addfile($filename);
+		$digest = $sha->hexdigest ;
+		if ( $digest ne $jars_info[$i]{sha1}) {
+			print "Expected sha1 is: $jars_info[$i]{sha1}, \n";
+			print "Actual sha1 is  : $digest. \n";
+			print "Please delete the existing one and download it again!";
+			die "ERRPR: $filename is broken, the sha1 sum is not invalid. \n";
+		}
```

Comment:
this is valid or invalid? :)

Reply:
It's invalid, thanks for pointing it out. Error message is updated.